### PR TITLE
MeanTimeReporter: shows only current running suites' times

### PR DESCRIPTION
**Pull request by Zhomart**

### CODE REVIEW

**Comment by os97673** [comment]
1. I expected to see an option for this functionality
2. it looks like we will print zeros for tests which were executed before but not in this run. Am I right?

**Comment by Zhomart** [comment]
1. I've added option `show_all_runs`, default is `false`.
2. I don't think we need to print previous tests results. It can be huge. Just for example, in my application I have ton's of tests. When I work on a single feature I usually run only one file and expect to see mean time results only for that test. 
3. With current mean time reporter implementation I see results of other projects too. I have to reset results every time. This PR fixes this too.

Examples of output of MeanTimeReporter:

```
$ rake

...................

Finished tests in 1.028275s, 18.4775 tests/s, 35.9826 assertions/s.

19 tests, 37 assertions, 0 failures, 0 errors, 0 skips

Minitest Reporters: Mean Time Report (Samples: 13, Order: :avg :desc)
Avg: 0.697630386  Min: 0.67746364   Max: 0.711190851  Last: 0.67746364   Description: MinitestReportersTest::ProgressReporterTest
Avg: 0.345755484  Min: 0.337173357  Max: 0.359638331  Last: 0.342493743  Description: MinitestReportersTest::JUnitReporterTest
Avg: 0.003980802  Min: 0.002855549  Max: 0.008773919  Last: 0.00375374   Description: MinitestReportersTest::ReportersTest
Avg: 0.003002799  Min: 0.001998129  Max: 0.004978362  Last: 0.002147834  Description: MinitestReportersTest::MinitestReporterPluginTest
Avg: 0.001391056  Min: 0.00056546   Max: 0.002035749  Last: 0.001236729  Description: MinitestReportersTest::SpecReporterTest
Avg: 0.000757966  Min: 0.000578746  Max: 0.00109078   Last: 0.000780946  Description: MinitestReportersTest::ExtensibleBacktraceFilterTest
```

```
$ ruby -Ilib:test test/unit/minitest/reporters_test.rb

......

Finished tests in 0.003574s, 1678.7025 tests/s, 1958.4862 assertions/s.

6 tests, 7 assertions, 0 failures, 0 errors, 0 skips

Minitest Reporters: Mean Time Report (Samples: 13, Order: :avg :desc)
Avg: 0.003936276  Min: 0.002734081  Max: 0.008773919  Last: 0.002734081  Description: MinitestReportersTest::ReportersTest
```

File: lib/minitest/reporters/mean_time_reporter.rb [hunk 1]
```diff
@@ -44,6 +44,8 @@ def self.reset_statistics!
       #   summary at the end of the test run. Default is 15.
       # @option show_progress [Boolean] If true it prints pass/skip/fail marks.
       #   Default is true.
+      # @option show_all_runs [Boolean] If true it shows all recorded suit results.
+      #   Default is false.
```
Comment (os97673) [inline lib/minitest/reporters/mean_time_reporter.rb]: I think default should be true, just to be compatible with previous behavior.

**Comment by Zhomart** [comment]
Default is true now.

**Pull request merged**